synapse/lib/storm.py

```diff
@@ -230,10 +230,32 @@
                     'perm': {'type': 'array', 'items': {'type': 'string'}},
                     'desc': {'type': 'string'},
                     'gate': {'type': 'string'},
+                    'workflowconfig': {'type': 'boolean'},
+                    'scopes': {
+                        'type': 'array',
+                        'items': {
+                            'type': 'string',
+                            'enum': ['global', 'self']
+                        },
+                    },
                 },
                 'required': ['perm', 'desc', 'gate'],
             },
         },
+        'configvars': {
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'name': {'type': 'string'},
```

Comment:
could we add some type knowledge ( either embedded JSON Schema or maybe cortex model types? ) to this so we can have auto normalization/enforcement?